cmd/metal-api/internal/service/machine-service.go

```diff
@@ -1143,6 +1147,51 @@ func createMachineAllocationSpec(ds *datastore.RethinkStore, machineRequest v1.M
 		return nil, fmt.Errorf("size:%s not found err:%w", sizeID, err)
 	}
 
+	partition, err := ds.FindPartition(partitionID)
+	if err != nil {
+		return nil, fmt.Errorf("partition:%s not found err:%w", partitionID, err)
+	}
+	var (
+		dnsServers metal.DNSServers
+		ntpServers metal.NTPServers
```

Comment:
This would render the `else` branches unnecessary:

```suggestion
		dnsServers = partition.DNSServers
		ntpServers = partition.NTPServers
```